src/sentry/integrations/jira/integration.py

```diff
@@ -207,12 +209,49 @@ def get_organization_config(self):
         client = self.get_client()
 
         try:
-            statuses = [(c["id"], c["name"]) for c in client.get_valid_statuses()]
-            configuration[0]["mappedSelectors"]["on_resolve"]["choices"] = statuses
-            configuration[0]["mappedSelectors"]["on_unresolve"]["choices"] = statuses
-
             projects = [{"value": p["id"], "label": p["name"]} for p in client.get_projects_list()]
             configuration[0]["addDropdown"]["items"] = projects
+
+            # We need to monitor if we're getting a large volume of requests
+            # with a significant number of projects. Issuing 5 requests or more
+            # per configuration load is something we may need to address via
+            # a bulk query.
```

Comment:
should we add custom exception handling if we get rate limited? it looks like the default behavior is to mark the config as disabled if we get any `ApiError` 😬

Reply:
Oh that's a really good point. Where's the `ApiError` exception handling that does this?

Reply:
https://github.com/getsentry/sentry/blob/3fd221e84d69ff3134d5a1abe95030f9470209c0/src/sentry/shared_integrations/client/base.py#L280
https://github.com/getsentry/sentry/blob/3fd221e84d69ff3134d5a1abe95030f9470209c0/src/sentry/shared_integrations/exceptions/__init__.py#L82-L91

looks like `ApiRateLimitedError`

Reply:
Oh, I was asking where the disable config behavior lives. I'm curious if this is something we'll want to narrow the exception handling for, since rate limiting should never disable an org config IMO.

Reply:
oh it's right under the changes you made for this PR
<img width="689" alt="Screenshot 2024-10-11 at 11 01 05" src="https://github.com/user-attachments/assets/b4bf2fb1-8f21-4a65-a54a-67908fb52fd0">

Reply:
Oh duh, it's in the same method. Never mind!

Reply:
Added logging and reraise the exception to show the user that we failed to load their configuration.

I suppose we could show them a partially filled in config, but that might let users misconfigure their Jira projects which I want to avoid